langworld_db_data/adders/listed_value_adder.py

```diff
@@ -16,54 +16,98 @@ def add_listed_value(
         new_value_en: str,
         new_value_ru: str,
         custom_values_to_rename: Optional[list[str]] = None,
+        index_to_assign: int = -1,
     ) -> None:
         """Adds listed value to the inventory and marks matching custom value
         in feature profiles as listed.  If one or more custom values
         in feature profiles were formulated differently but now
         have to be renamed to be the new `listed` value, these custom values
         can be passed as a list.
+        If no id to assign is given, the new value will be appended after
+        the last present value, else the new value will receive ID based on the passed index.
         """
+
         if not (feature_id and new_value_en and new_value_ru):
             raise ListedValueAdderError("None of the passed strings can be empty")
 
-        id_of_new_value = self._add_to_inventory_of_listed_values(
-            feature_id=feature_id, new_value_en=new_value_en, new_value_ru=new_value_ru
-        )
-        self._mark_value_as_listed_in_feature_profiles(
-            feature_id=feature_id,
-            new_value_id=id_of_new_value,
-            new_value_ru=new_value_ru,
-            custom_values_to_rename=custom_values_to_rename,
-        )
+        try:
+            id_of_new_value = self._add_to_inventory_of_listed_values(
+                feature_id=feature_id,
+                new_value_en=new_value_en,
+                new_value_ru=new_value_ru,
+                index_to_assign=index_to_assign,
+            )
+
+            self._mark_value_as_listed_in_feature_profiles(
+                feature_id=feature_id,
+                new_value_id=id_of_new_value,
+                new_value_ru=new_value_ru,
+                custom_values_to_rename=custom_values_to_rename,
+            )
+        except ValueError as e:
+            raise ListedValueAdderError(e)
 
     def _add_to_inventory_of_listed_values(
         self,
         feature_id: str,
         new_value_en: str,
         new_value_ru: str,
+        index_to_assign: int,
     ) -> str:
+        """
+        Returns ID of new value and tuple of indices that must be incremented in feature profiles.
+        index_to_assign means number that must be assigned to the new value within the feature, ex. 13 for A-2-13.
+        If no index_to_assign is given, the new value will be added as the last one in the feature.
+        index_to_assign must be greater than 0.
+        """
+
         rows = read_dicts_from_csv(self.input_file_with_listed_values)
 
         if not [r for r in rows if r["feature_id"] == feature_id]:
             raise ListedValueAdderError(f"Feature ID {feature_id} not found")
 
-        index_of_last_row_for_given_feature = 0
-        id_of_new_value = ""
+        # Collect all indices and line numbers of given feature into value_indices_to_inventory_line_numbers
```

Comment:
```suggestion
        # Collect all indices and line numbers of given feature
```
no need to repeat variable name IMO. Also, as you can see, the comment totally repeats the name of the method being called. It may still make sense to have a comment here, though, but a comment explaining _why_ we're doing this, not _what_ we're doing. See [CSI](https://standards.mousepawmedia.com/en/latest/csi.html) principle.

I think the comment could include: what indices exactly we're getting (for the feature that we're inserting a new value into) and why we need this dictionary

Reply:
Done!